src/main/scala/com/codacy/client/stash/service/RepositoryServices.scala

```diff
@@ -1,30 +1,89 @@
 package com.codacy.client.stash.service
 
-import com.codacy.client.stash.client.{Request, RequestResponse, StashClient}
-import com.codacy.client.stash.{Repository, SshKeySimple}
+import com.codacy.client.stash.client.{PageRequest, Request, RequestResponse, StashClient}
+import com.codacy.client.stash.{Repository, SshKeySimple, UserPermission}
 import play.api.libs.json.Json
 
 class RepositoryServices(client: StashClient) {
 
-  /*
-   * Gets the list of the user's repositories. Private repositories only appear on this list
-   * if the caller is authenticated and is authorized to view the repository.
-   */
-  def getRepositories(projectKey: String): RequestResponse[Seq[Repository]] = {
-    client.executePaginated(Request(s"/rest/api/1.0/projects/$projectKey/repos", classOf[Seq[Repository]]))
+  val BASE: String = "/rest/api/1.0/projects"
+
+  /**
+    * Gets the list of the user's repositories. Private repositories only appear on this list
+    * if the caller is authenticated and is authorized to view the repository.
+    */
+  def getRepositories(projectKey: String, pageRequest: Option[PageRequest]): RequestResponse[Seq[Repository]] =
```

Comment:
since we're changing this function signature, this should be breaking and we should add something on the changelog